File: abstractions/dotnet/src/store/BackingStoreFactorySingleton.cs
// ------------------------------------------------------------------------------
//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
// ------------------------------------------------------------------------------

namespace Microsoft.Kiota.Abstractions.Store
{
    /// <summary>
    ///     This class is used to register the backing store factory.
    /// </summary>
    public class BackingStoreFactorySingleton
    {
        /// <summary>
        ///     The backing store factory singleton instance.
        /// </summary>
        public static IBackingStoreFactory Instance { get; set; } = new InMemoryBackingStoreFactory();
    }
}

File: abstractions/php/src/Store/BackingStoreFactorySingleton.php
<?php

namespace Microsoft\Kiota\Abstractions\Store;

abstract class BackingStoreFactorySingleton
{
    /**
     * @var BackingStoreFactory|null
     */
    public static ?BackingStoreFactory $instance = null;

    /**
     * We use the getter method since PHP doesn't support instantiating an instance
     * outside a method.
     * @return BackingStoreFactory|null
     */
    public static function getInstance(): ?BackingStoreFactory {
        if (is_null(self::$instance)) {
            self::$instance = new InMemoryBackingStoreFactory();
        }
        return self::$instance;
    }
}